src/util/combo.test.js

```diff
@@ -275,5 +275,69 @@ describe("src/util/combo", () => {
         expect(p.parseOrThrow(v)).toEqual(v);
       });
     });
+    describe("with field renaming", () => {
+      const p: C.Parser<{|
+        +one: number,
+        +two: number,
+        +three?: number,
+        +four?: number,
+      |}> = C.object(
+        {one: C.number, two: C.rename("dos", C.number)},
+        {three: C.number, four: C.rename("cuatro", C.number)}
+      );
+      it("renames both required and optional fields", () => {
+        expect(p.parseOrThrow({one: 1, dos: 2, three: 3, cuatro: 4})).toEqual({
+          one: 1,
+          two: 2,
+          three: 3,
+          four: 4,
+        });
+      });
+      it("provides missing key errors using the user-facing name", () => {
+        const thunk = () => p.parseOrThrow({one: 1, cuatro: 4});
+        expect(thunk).toThrow('missing key: "dos"');
+      });
+      it("only accepts the user-facing keys", () => {
+        const thunk = () => p.parseOrThrow({one: 1, two: 2});
+        expect(thunk).toThrow('missing key: "dos"');
+      });
+      it("only accepts the user-facing keys", () => {
```

Comment:
Duplicated test.

Reply:
Thanks—I think that I meant to make this a test for user-facing optional
keys, which I’ve now done.